src/main/java/com/SAS/crudoperations/LeagueCRUD.java

```diff
@@ -1,98 +1,186 @@
+            } catch (EmptyResultDataAccessException e) {
+                return false;
+            }
+        } else {
+            return false;
+        }
+    }
 
-    public void addLeagueScore(int league, Season season) {
-
+    public static boolean removeLeague(String name) {
+        try {
+            String query = String.format("DELETE FROM league WHERE league_name=\"%s\"", name);
+            jdbcTemplate.update(query);
+            return true;
+        } catch (EmptyResultDataAccessException e) {
+            return false;
+        }
     }
 
-    public int getSeason(int year) {
+    public static boolean removeSeason(Season season) {
         try {
-            String query = String.format("SELECT season_id FROM season WHERE year=" + year);
-            int id = jdbcTemplate.queryForObject(query, int.class);
-            return id;
+            String query = String.format("DELETE FROM season WHERE season_year=\"%d\"", season.getYear());
+            jdbcTemplate.update(query);
+            return true;
         } catch (EmptyResultDataAccessException e) {
-            return -1;
+            return false;
         }
     }
 
-    public boolean isRefExist(int refID) {
+    public static boolean isRefExist(Referee ref) {
         try {
-            String query = String.format("SELECT * FROM referee WHERE referee_id=" + refID);
+            String query = String.format("SELECT user_id FROM referee WHERE user_id=\"%d\"", UsersCRUD.getUserIdByUserName(ref.getUserName()));
+            if (UsersCRUD.getUserIdByUserName(ref.getUserName()) == -1) {
+                return false;
+            }
             jdbcTemplate.queryForObject(query, int.class);
             return true;
         } catch (EmptyResultDataAccessException e) {
             return false;
         }
     }
 
-    public void removeRefFromLeague(Referee ref) {
-        if(isRefExist(ref.getUser().getUserID())){
-            String query = String.format("DELETE FROM league_referees WHERE referee_id=" + ref.getUser().getUserID());
-            jdbcTemplate.update(query, int.class);
+    public static boolean removeRefFromLeague(Referee ref, League league) {
+        try {
+            String query = String.format("DELETE FROM referees_in_duty WHERE user_id=%d AND league_name=\"%s\"", ref.getUserID(), league.getName());
+            if(jdbcTemplate.update(query)==0||ref.getUserID()==-1){
+                return false;
+            }
+            return true;
+        } catch (EmptyResultDataAccessException e) {
+            return false;
         }
     }
 
-    public void addRefToLeague(Referee ref) {
-        if(!isRefExistInLeague(ref.getUser().getUserID())){
-            String query = String.format("insert into league_referees values (%d, %d);", ref.getUser().getUserID(),ref.getLevel() );
-            jdbcTemplate.update(query, int.class);
+    public static boolean addRefToLeagueInSeason(League league, Season season, HashSet<Referee> referees) {
```

Comment:
Why do you get all the referees (and should get strings like refName and pull its matching id from db) instead of just adding one referee here and call this function for each referee you want to add?

Reply:
the UC as i understood it is to add a couple of referees and not only one.
thats way im using the HashMap, because there is an attribute of a HashMap of referees in sesason.